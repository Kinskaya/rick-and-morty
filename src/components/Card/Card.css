.card-container {
    display: flex;
    flex-wrap: wrap;
    margin-top: 32px;
}

.card {
    width: 222px;
    height: 270px;
    background: radial-gradient(109.56% 109.56% at 0% -2.94%, rgba(255, 255, 255, 0.5) 0%, rgb(217, 208, 248) 100%);
    backdrop-filter: blur(4px);
    border-radius: 10px;
    padding-left: 16px;
    padding-top: 18px;
    margin: 0 0 36px 41px;
}

.card p {
    margin: 0 0 5px 0;
}
.card-image {
    height: 107px;
    object-fit: contain;
}

.card-add-info {
    display: none;
}

.title {
    color: #414191;
}

.title span {
    color: #FC466B;
}
