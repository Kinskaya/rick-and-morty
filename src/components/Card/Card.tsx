import React from 'react';
import { useNavigate } from 'react-router-dom';

import './Card.css';

export interface ICard {
  id: number;
  name: string;
  gender: string;
  image: string;
  status: string;
  species: string;
  created: string;
  origin: {
    name: string;
  };
}

const Card: React.FunctionComponent<ICard> = ({ ...item }) => {
  const router = useNavigate();

  return (
    <div className="card" onClick={() => router(`/${item.id}`)}>
      <img className="card-image" src={item.image} alt="item-image" />
      <p className="title">
        Name: <span>{item.name}</span>
      </p>
      <p className="title">
        Gender: <span>{item.gender}</span>
      </p>
    </div>
  );
};

export default Card;
