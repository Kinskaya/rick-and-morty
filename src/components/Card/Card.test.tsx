import { render } from '@testing-library/react';
import Card from './Card';

describe('Card component', () => {
  it('render correctly', () => {
    const item = {
      id: 1,
      name: 'MacBook',
      price: 1400,
      imageUrl: 'https://picsum.photos/id/180/2400/1600',
    };
    const { asFragment } = render(
      <Card id={item.id} imageUrl={item.name} name={item.name} price={item.price} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
