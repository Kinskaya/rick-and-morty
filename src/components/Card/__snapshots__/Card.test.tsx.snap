// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component render correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="item-image"
      class="card-image"
    />
    <p
      class="title"
    >
      Name: 
      <span>
        MacBook
      </span>
    </p>
    <p
      class="title"
    >
      Gender: 
      <span />
    </p>
  </div>
</DocumentFragment>
`;
